comp/metadata/packagesigning/packagesigningimpl/yumsigning.go

```diff
@@ -0,0 +1,185 @@
+		}
+	}
+	// Then parsing of the repo files
+	if _, err := os.Stat(repoFilesDir); os.IsExist(err) {
+		if files, err := os.ReadDir(repoFilesDir); err == nil {
+			for _, file := range files {
+				_, gpgFiles = parseRepoFile(file.Name(), gpgCheck)
+				for _, gpgFile := range gpgFiles {
+					decryptGPGFile(allKeys, gpgFile, "repo")
+				}
+			}
+		}
+	}
+}
+
+// getRepoPathFromPkgManager returns the path to the configuration file and the path to the repository files for RH or SUSE based OS
+func getRepoPathFromPkgManager(pkgManager string) (string, string) {
+	if pkgManager == "zypper" {
+		return zyppConf, zyppRepo
+	}
+	return yumConf, yumRepo
+}
+
+func getKeysFromRPMDB(allKeys map[SigningKey]struct{}) {
+	// It seems not possible to get the expiration date from rpmdb, so we extract the list of keys and call gpg
+	cmd := exec.Command("rpm", "-qa", "gpg-pubkey*")
+	output, err := cmd.Output()
+	if err != nil {
+		return
+	}
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+	for scanner.Scan() {
+		rpmCmd := exec.Command("rpm", "-qi", "%s", "--qf", "'%%{PUBKEYS}\n'")
+		rpmKey, err := rpmCmd.Output()
+		if err != nil {
+			return
+		}
+		decryptGPGReader(allKeys, strings.NewReader(string(rpmKey)), "rpm")
+	}
+}
+
+// parseRepoFile extracts information from yum repo files
+// Save the global gpgcheck value when encountering a [main] table (should only occur on `/etc/yum.conf`)
+// Match several entries in gpgkey field, either file references (file://) or http(s)://. From observations,
+// these reference can be separated either by space or by new line. We assume it possible to mix file and http references
+func parseRepoFile(inputFile string, gpgConf bool) (bool, []string) {
+
+	file, err := os.Open(inputFile)
+	if err != nil {
+		return true, nil
+	}
+	defer file.Close()
+
+	signedBy := regexp.MustCompile(`file://([A-Za-z0-9_\-\.\/]+)`)
+	table := regexp.MustCompile(`\[[A-Za-z0-9_\-\.\/ ]+\]`)
+	urlmatch := regexp.MustCompile(`http(s)?://[A-Za-z0-9_\-\.\/]+`)
+	gpgCheck := gpgConf
+	localGpgCheck := false
+	inMain := false
+	var gpgFiles []string
+	var localGpgFiles []string
+	defer file.Close()
+	if err != nil {
+		return true, nil
+	}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		currentTable := table.FindString(line)
+		if currentTable == "[main]" {
+			inMain = true
+		} else if currentTable != "" {
+			// entering new table, save values
+			if localGpgCheck {
+				gpgFiles = append(gpgFiles, localGpgFiles...)
+			}
+			// reset
+			inMain = false
+			localGpgCheck = gpgCheck
+			localGpgFiles = nil
+		}
+		if strings.HasPrefix(line, "gpgcheck") {
+			if inMain {
+				gpgCheck = strings.Contains(line, "1")
+			} else {
+				localGpgCheck = strings.Contains(line, "1")
+			}
+		}
+		if strings.HasPrefix(line, "gpgkey") {
+			// check format `gpgkey=file///etc/file1 file:///etc/file2`
+			signedFile := signedBy.FindAllStringSubmatch(line, -1)
```

Comment:
Nit: should this variable be called `pubkeyURI`? First of all it's not file, it's an URI (pointing to a local file or http(s) hosted file) and second of all it's not "signed", it's a public key path.